src/modules/dashboard/components/Extensions/ExtensionStatus.tsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import { defineMessages } from 'react-intl';
+
+import Tag from '~core/Tag';
+
+import { ColonyExtension } from '~data/index';
+
+const MSG = defineMessages({
+  installed: {
+    id: 'dashboard.Extensions.ExtensionStatus.installed',
+    defaultMessage: 'Installed',
+  },
+  notInstalled: {
+    id: 'dashboard.Extensions.ExtensionStatus.notInstalled',
+    defaultMessage: 'Not installed',
+  },
+  missingPermissions: {
+    id: 'dashboard.Extensions.ExtensionStatus.missingPermissions',
+    defaultMessage: 'Missing permissions',
+  },
+  deprecated: {
+    id: 'dashboard.Extensions.ExtensionStatus.deprecated',
+    defaultMessage: 'Deprecated',
+  },
+  enabled: {
+    id: 'dashboard.Extensions.ExtensionStatus.enabled',
+    defaultMessage: 'Enabled',
+  },
+  notEnabled: {
+    id: 'dashboard.Extensions.ExtensionStatus.notEnabled',
+    defaultMessage: 'Not enabled',
+  },
+});
+
+interface Props {
+  deprecatedOnly?: boolean;
+  installedExtension?: ColonyExtension | null;
+}
+
+const ExtensionStatus = ({
+  deprecatedOnly = false,
+  installedExtension,
+}: Props) => {
+  let status;
+  let theme;
+
+  if (!installedExtension) {
+    status = MSG.notInstalled;
+  } else if (!installedExtension.details.initialized) {
+    status = MSG.notEnabled;
+    theme = 'golden';
+  } else if (installedExtension.details.missingPermissions.length) {
+    status = MSG.missingPermissions;
+    theme = 'danger';
+  } else if (installedExtension.details.initialized) {
+    status = MSG.enabled;
+    theme = 'primary';
+  } else {
+    status = MSG.installed;
+  }
+  return (
+    <>
+      {!deprecatedOnly ? <Tag appearance={{ theme }} text={status} /> : null}
```

Comment:
What will happen if we pass undefined value for theme?

Reply:
With the theme set to undefined, it will use the base `main` class styles, which will render like this:

![Screenshot from 2021-02-23 20-53-38](https://user-images.githubusercontent.com/1193222/108893082-522ef080-7619-11eb-936e-40528453a4fc.png)